### DISCUSSION TITLE
How to specify ESNext (uncompiled) entry for an npm package?

### DISCUSSION THREAD

**Discussion by HosseinAgha** [vercel/next.js#30451] [Help]
TLDR, I want my package's package.json to be something like this:  

```json
{
  "name": "",
  "main": "index.js",
  "module": "index.mjs",
  "esnext": "index.esnext.mjs"
}
```

I know there is no such a thing as `esnext` field in package.json, this is rhetorical.   

I want the [modern bundle](https://nextjs.org/blog/next-9-1#module--nomodule) of the Next.js to load the esnext (untranspiled) version of my package.  
The module field is a transpiled es module build by convention (as no one would transpile package.json packages)

The issue here is that there is no way to publish a fully ESNext entrypoint for the package in order for Next.js to use for its modern bundle. The result is that although Next.js won't compile the modern bundles but the package's code is already compiled to the legacy IE11 code and this makes this optimization ineffective for the external packages.

Do you know any solutions to make this happen?
